### ISSUE TITLE
_delete only for GPU ?

### ISSUE THREAD

**Issue by Lecrapouille** [glumpy/glumpy#218]
Hi ! This is probably not a big issue but because `def delete(self):` is public and calls `def _delete(self):` should not we clear nuppy array as well as internal states for Texture2D, Buffer, FrameBuffer?

For example:
- texture.py: should set width and height to 0, reset pending_data, and nuppy array to 0.
- framebuffer.py: should set _color, _depth ... to None.
- buffer.py: should reset pending_data, and nuppy array to 0 ...
etc ?

**Comment by rougier** [comment]
I'm not sure I get the context where this happens. Could you develop a bit?

**Comment by Lecrapouille** [comment]
Hi thank for your reply! Sorry, I was a little tired when I wrote this comment and I'm a very beginner in Python so it's more a hypothesis than concrete facts. I rephrase:
 
Let focus on Texture2D to stay simple.

Currently, if I call `delete` this method will call `delete_`  which gonna clean GPU memory and reset GLObject states such as `_need_setup` ... As a consequence, no more texture held inside OpenGL. This behavior is fine for me. 

But now suppose I then call `width` or `heigh` this will return the older values instead of reset values (0 or None). Same idea with CPU/Numpy data holding the older texture bytes. Should not be safer to reset the array size to 0 ? Same idea for pending data holding older states. Should not reset them to None? Logically, after `delete` if I try to add new textures bytes, they will be appended to older and pending data will force transferring to GPU the numpy array with older and newer bytes.

If my hypothesis is correct, we can suppose the same problem for framebuffer, VBO ...

**Comment by rougier** [comment]
I see your point now. It make sense but in the meantime, if you delete a texture, you're not supposed to access it anymore, no ?

**Comment by Lecrapouille** [comment]
> I see your point now. It make sense but in the meantime, if you delete a texture, you're not supposed to access it anymore, no ?

Yes this totally makes sense to not access to it anymore, but I was thinking about GLObject states such as `_need_setup, _need_create` which can potentially create internal structure back (kind of scenario where you release your texture then create it back/re-use later just by setting new data on it)? But I guess this scenario happens rarely.

Again, it was more a remark than a real issue! So feel free to close this ticket! :)

**Comment by rougier** [comment]
Sorry for  very late answer. Can you make a PR with your proposed changes ?

**Comment by Lecrapouille** [comment]
No problem :) Sorry for the PR I hardly know Python, so I'm not feeling comfortable to create a PR  potential buggy so maybe I can post prototype of changes?

**Comment by rougier** [comment]
ok
